lib/apiservers/portlayer/restapi/handlers/storage_handlers.go

```diff
@@ -350,6 +357,38 @@ func (h *StorageHandlersImpl) WriteImage(params storage.WriteImageParams) middle
 	return storage.NewWriteImageCreated().WithPayload(i)
 }
 
+//ImageJoin modifies the config spec of a container to include the specified image
+func (h *StorageHandlersImpl) ImageJoin(params storage.ImageJoinParams) middleware.Responder {
+	op := trace.NewOperation(context.Background(), "ImageJoin %s", params.ID)
+	defer trace.End(trace.Begin("", op))
+
+	handle := epl.HandleFromInterface(params.Config.Handle)
+	if handle == nil {
+		err := &models.Error{Message: "Failed to get the Handle"}
+		return storage.NewImageJoinInternalServerError().WithPayload(err)
+	}
+
+	storeURL, _ := util.ImageStoreNameToURL(params.StoreName)
+	img, err := h.imageCache.GetImage(op, storeURL, params.ID)
+	if err != nil {
+		op.Errorf("Volumes: StorageHandler : %#v", err)
+		return storage.NewImageJoinNotFound().WithPayload(&models.Error{Code: http.StatusNotFound, Message: err.Error()})
+	}
+
+	cfg := params.Config
+	handleprime, err := vsimage.Join(op, handle, cfg.DeltaID, cfg.ImageID, cfg.RepoName, img)
+	if err != nil {
+		op.Errorf("join image failed: %#v", err)
+		return storage.NewImageJoinInternalServerError().WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	op.Debugf("image %s has been joined to %s as %s", params.ID, handle.Spec.ID(), cfg.DeltaID)
```

Comment:
This could be useful at the `Info` level, it is an important part of the container create cycle.